src/string_lib.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+
+size_t str_len(const char *s)
+{
+  if (NULL == s)
+  {
+    return 0;
+  }
+
+  size_t length = 0;
+  while (*s != 0)
+  {
+    length++;
+    s++;
+  }
+
+  return length;
+}
+
+int str_cmp(const char *s1, const char *s2)
+{
+  int res = 0;
+
+  while (*s1 != 0 && *s2 != 0)
```

Comment:
```suggestion
  while (*s1 != 0 && *s2 != 0 && *s1 == *s2)
```
You need to compare the strings as you go along!

Reply:
@tony-go You are still not comparing the strings here!